File: water/water.java
public class water {
	static int max(int a,int b){
		if(a>b){
			return a;
		}else{
			return b;
		}	
	}
	static int min(int a,int b){
		if(a<b){
			return a;
		}else{
			return b;
		}
	}

	public static void main(String[] args){
		System.out.println("unko");
		int a = 100;
		int i,j;
		System.out.println(a);

		//ここからメインの計算
		for(i=0;i<numberi-1;++i){
			for(j=i+1;j<number;++j){
				volume = max((min(b[i],b[j])*(j-i)),volume);
			}
		}
		System.out.println("maximam is" volume);
	}
}

File: water/water.pl
#!/usr/bin/perl
use strict;
use warnings;

my (@high)= @ARGV;
my $num = @ARGV;
print "@high\n";

my $number = $high[0];

if ($num != $number + 1){
	print "The number you entered and the number of pole is not the same\n";
	exit;
}
my $volume = 0;
for (my $i = 1;$i<$number;$i++){
	for (my $j = 2;$j<$number+1;$j++){
		&max(&min($high[$i],$high[$j])*($j-$i),$volume);
	}
}
print $volume;

sub max{
	(my $x,my $y) = @_;
	if($x>$y){
		return $x;
	}else{
		return $y;
	}
}
sub min{		
	(my $x,my $y) = @_;
	if($x<$y){
		return $x;
	}else{
		return $y;
	}
}
